homeassistant/components/climate/tahoma.py

```diff
@@ -0,0 +1,73 @@
+"""
+Support for Tahoma heatpump derogation.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.tahoma/
+"""
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+    SUPPORT_AWAY_MODE)
+from homeassistant.components.tahoma import (
+    DOMAIN as TAHOMA_DOMAIN, TahomaDevice)
+from homeassistant.const import TEMP_CELSIUS
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Tahoma controller devices."""
+    controller = hass.data[TAHOMA_DOMAIN]['controller']
+    devices = []
+    for device in hass.data[TAHOMA_DOMAIN]['devices']['climate']:
+        devices.append(TahomaClimate(device, controller))
+    add_devices(devices, True)
+
+
+class TahomaClimate(TahomaDevice, ClimateDevice):
+    """Representation of a Tahoma heat pump derogration device."""
+
+    def __init__(self, tahoma_device, controller):
+        """Initialize the climate device."""
+        super().__init__(tahoma_device, controller)
+        if self.tahoma_device.type == \
+           'io:EnergyConsumptionSensorsHeatPumpComponent':
+            self._name = 'Heat pump derogation ('+self.name+')'
```

Comment:
Please use string formatting. I'm not sure if we are using bracket somewhere for the name.

Reply:
Could you develop that ? I could totally remove the sensor name wich is really non human.
Or another presentation?

Reply:
done